parsers/IN_WE.py

```diff
@@ -93,17 +98,46 @@ def fetch_data(
                 item[datetime_col] = dt.shift(minutes=1).floor("minute").datetime
             else:
                 item[datetime_col] = dt.floor("minute").datetime
-        df_data = pd.DataFrame(
-            [item for item in data if item[datetime_col].hour == dt_12_hour.hour]
-        )
-        return df_data
+        return data
     else:
         raise ParserException(
             parser="IN_WE.py",
             message=f"{target_datetime}: {kind} data is not available",
         )
 
 
+def format_raw_data(
+    kind: str,
+    data: json,
+    zone_key: str,
+    target_datetime: datetime,
+    zone_key2: Optional[str] = None,
+) -> dict:
+    assert data is not None
+    assert kind is not None
+
+    dt_12_hour = arrow.get(target_datetime.strftime("%Y-%m-%d %I:%M")).datetime
+    datetime_col = KIND_MAPPING[kind]["datetime_column"]
+    df_data = pd.DataFrame(
+        [item for item in data if item[datetime_col].hour == dt_12_hour.hour]
+    )
+    if kind == "production":
```

Comment:
I'm thinking we could do a mapping here but I'm not sure it would improve the readability of the code.
Something like:
```
FORMAT_FUNCTIONS = {
"production": format_production_data,
"consumption": format_consumption_data,
"exchange": format_exchanges_data
}
formatted_data_point = FORMAT_FUNCTIONS[kind](**kwargs)
```
But might make the code more complex to read and there's still some logic to do for exchange anyways.